src/frontend/citizen-portal/src/app/core/services/form-utils.service.ts

```diff
@@ -0,0 +1,109 @@
+import { Injectable } from '@angular/core';
+import {
+  AbstractControl,
+  FormGroup,
+  FormControl,
+  ValidatorFn,
+  FormArray,
+  FormBuilder,
+} from '@angular/forms';
+
+import { LoggerService } from './logger.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class FormUtilsService {
+  constructor(private fb: FormBuilder, private logger: LoggerService) {}
+
+  /**
+   * @description
+   * Checks the validity of a form, and triggers validation messages when invalid.
+   */
+  public checkValidity(form: FormGroup | FormArray): boolean {
+    if (form.valid) {
+      return true;
+    } else {
+      this.logger.info('FORM_INVALID', this.getFormErrors(form));
+
+      form.markAllAsTouched();
+      return false;
+    }
+  }
+
+  /**
+   * @description
+   * Sets FormControl validators.
+   */
+  public setValidators(
+    control: FormControl | FormGroup,
+    validators: ValidatorFn | ValidatorFn[],
+    blacklist: string[] = []
+  ): void {
+    if (control instanceof FormGroup) {
+      // Assumes that FormGroups will not be deeply nested
+      Object.keys(control.controls).forEach((key: string) => {
+        // Skip blacklisted keys from having validators updated
+        if (!blacklist.includes(key)) {
```

Comment:
Try to avoid using terms blacklist / whitelist. These terms are considered [oppressive language](https://tools.ietf.org/id/draft-knodel-terminology-00.html#rfc.section.1.2). Recommended alternative language could be blockedKeys or something similar.

Reply:
Also see the repo-mounti issue indicating the same information https://github.com/bcgov/jag-traffic-courts-online/issues/1